identity-apps-core/apps/recovery-portal/src/main/webapp/includes/init-url.jsp

```diff
@@ -0,0 +1,169 @@
+    String commonauthURL = "../commonauth";
+    String oauth2AuthorizeURL = "../oauth2/authorize";
+    String oidcLogoutURL = "../oidc/logout";
+    String openidServerURL = "../openidserver";
+    String logincontextURL = "../logincontext";
+    String longwaitstatusURL = "/longwaitstatus";
+
+    String tenantDomain;
+    String userTenantDomain;
+    String tenantForTheming;
+    String userTenant;
+    String spAppName;
+    String spAppId;
+
+    if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+        tenantDomain = IdentityTenantUtil.resolveTenantDomain();
+        tenantForTheming = tenantDomain;
+        userTenant = tenantDomain;
+
+        spAppName = request.getParameter(SERVICE_PROVIDER_NAME_SHORT);
+        if (StringUtils.isBlank(spAppName) && StringUtils.isNotBlank((String) request.getAttribute(SERVICE_PROVIDER_NAME_SHORT))) {
+            spAppName = (String) request.getAttribute(SERVICE_PROVIDER_NAME_SHORT);
+        }
+
+        spAppId = request.getParameter(SERVICE_PROVIDER_ID_SHORT);
+        if (StringUtils.isBlank(spAppId) && StringUtils.isNotBlank((String) request.getAttribute(SERVICE_PROVIDER_ID_SHORT))) {
+            spAppId = (String) request.getAttribute(SERVICE_PROVIDER_ID_SHORT);
+        }
+
+        String tenantDomainFromURL = request.getParameter(TENANT_DOMAIN_SHORT);
+        if (StringUtils.isBlank(tenantDomainFromURL) && StringUtils.isNotBlank((String) request.getAttribute(TENANT_DOMAIN_SHORT))) {
+            tenantDomainFromURL = (String) request.getAttribute(TENANT_DOMAIN_SHORT);
+        }
+
+        String tenantDomainOfUser = request.getParameter(USER_TENANT_DOMAIN_SHORT);
+        if (StringUtils.isBlank(tenantDomainOfUser) && StringUtils.isNotBlank((String) request.getAttribute(USER_TENANT_DOMAIN_SHORT))) {
+            tenantDomainOfUser = (String) request.getAttribute(USER_TENANT_DOMAIN_SHORT);
+        }
+
+        userTenantDomain = tenantDomainOfUser;
+
+        if (StringUtils.equals(tenantDomain, IdentityManagementEndpointConstants.SUPER_TENANT)
+            && StringUtils.equals(spAppName, "Console")) {
+            tenantForTheming = IdentityManagementEndpointConstants.SUPER_TENANT;
+        } else {
+            if (StringUtils.isBlank(userTenantDomain)) {
+                userTenantDomain = tenantDomainFromURL;
+            }
+            if (StringUtils.isBlank(userTenantDomain)) {
+                userTenantDomain = tenantDomain;
+            }
+            if (StringUtils.isNotBlank(tenantDomainOfUser)) {
+                tenantForTheming = tenantDomainOfUser;
+            }
+            if (StringUtils.equals(spAppName, "My Account")
+                && StringUtils.equals(tenantDomain, IdentityManagementEndpointConstants.SUPER_TENANT)) {
+                tenantForTheming = userTenantDomain;
+                userTenant = userTenantDomain;
+            }
+        }
+    } else {
+        tenantDomain = request.getParameter(TENANT_DOMAIN_KEY);
```

Comment:
Let's encode these values to avoid code injection. Please check and fix the other places as well.

Reply:
Addressed with https://github.com/wso2/identity-apps/pull/5817/commits/877baa435b1c4ba3cfffef80cd1e5a56dc36d0c6